### PR TITLE
[authmanager] Fix OAuth2 implicit grant flow

### PR DESCRIPTION
**Pull request by elpaso**
Fix an unreported issue with OAuth2 implicit grant flow.

### CODE REVIEW

File: src/auth/oauth2/qgso2.cpp [hunk 1]
```diff
@@ -237,6 +237,15 @@ void QgsO2::link()
   }
 }
 
+
+void QgsO2::setState( const QString & )
```
Comment (dakcarto) [inline src/auth/oauth2/qgso2.cpp]: Should this have `const QString &value`. I mean, is `state` no longer user-configurable, or should it always just be a random number?

**Comment by dakcarto** [comment]
Looks OK to me. I guess the OGC folks must have forked prior to the implicit grant flow additions I made. However, note that I already made those same changes in [external/o2/src/o2.cpp](https://github.com/qgis/QGIS/blob/master/external/o2/src/o2.cpp).

What is the reasoning behind moving all that code up into `src/auth/oauth2/qgso2.cpp`? To use older `o2` libraries instead of the `external` one in src?

File: src/auth/oauth2/qgso2.cpp (hunk 1 above)
Comment (elpaso) [reply src/auth/oauth2/qgso2.cpp]: correct: it's no longer user configurable: it's a nonce opaque value, automatically generated

**Comment by nyalldawson** [comment]
@elpaso as discussed, tested and works well here.

**Comment by elpaso** [comment]
Thanks!

**Pull request merged**